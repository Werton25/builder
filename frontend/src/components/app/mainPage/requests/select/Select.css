.request-select .Select-menu-outer {
    z-index: 5;
}