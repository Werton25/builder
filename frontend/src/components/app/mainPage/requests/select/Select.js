import React from 'react';
import PropTypes from 'prop-types';
import SelectOption from "./SelectOption";
import {getCurrentRequest, getCurrentRequestsValues} from "../../../../../selectors/requests";
import {changeSelected} from "../../../../../actions/select";
import {connect} from 'react-redux';
import ReactSelect from 'react-select';
import 'react-select/dist/react-select.css';
import './Select.css'
import {getCurrentUri} from "../../../../../selectors/router";

export class BaseSelect extends React.Component {
  render() {
    const {currentUri, currentRequest, currentRequests, changeSelected} = this.props;
    return (
      <ReactSelect className='request-select'
                        autosize={false}
                        options={currentRequests}
                        clearable={false}
                        value={currentRequest}
                        labelKey='name'
                        onChange={(selected) => changeSelected(currentUri, selected.id)}
                        optionComponent={SelectOption}/>
    );
  }
}

BaseSelect.propTypes = {
  currentUri: PropTypes.string.isRequired,
  currentRequest: PropTypes.object.isRequired,
  currentRequests: PropTypes.array.isRequired,
  changeSelected: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  currentUri: getCurrentUri(state),
  currentRequest: getCurrentRequest(state),
  currentRequests: getCurrentRequestsValues(state)
});

const mapDispatchToProps = {
  changeSelected
};

export default connect(mapStateToProps, mapDispatchToProps)(BaseSelect);
