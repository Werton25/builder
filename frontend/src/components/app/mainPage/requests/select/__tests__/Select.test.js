import React from "react";
import renderer from 'react-test-renderer';
import {BaseSelect} from "../Select";
jest.mock('../SelectOption');

describe('Select tests', () => {
  it('Select renders correctly', () => {
    const tree = renderer
      .create(
          <BaseSelect currentUri={''} currentRequest={{}} currentRequests={[]} changeSelected={jest.fn()}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
