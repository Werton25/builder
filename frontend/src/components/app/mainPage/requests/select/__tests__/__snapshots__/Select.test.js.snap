// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select tests Select renders correctly 1`] = `
<div
  className="Select request-select has-value is-searchable Select--single"
  style={undefined}
>
  <div
    className="Select-control"
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <span
      className="Select-multi-value-wrapper"
      id="react-select-2--value"
    >
      <div
        className="Select-value"
        style={undefined}
        title={undefined}
      >
        <span
          aria-selected="true"
          className="Select-value-label"
          id="react-select-2--value-item"
          role="option"
        />
      </div>
      <div
        className="Select-input"
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <input
          aria-activedescendant="react-select-2--value"
          aria-describedby={undefined}
          aria-expanded="false"
          aria-haspopup="false"
          aria-label={undefined}
          aria-labelledby={undefined}
          aria-owns=""
          className="Select-input"
          id={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          role="combobox"
          tabIndex={undefined}
          value=""
        />
      </div>
    </span>
    <span
      className="Select-arrow-zone"
      onMouseDown={[Function]}
    >
      <span
        className="Select-arrow"
        onMouseDown={[Function]}
      />
    </span>
  </div>
</div>
`;
